### DISCUSSION TITLE
How to uze zrepl on an existing replacated set of snapshots

### DISCUSSION THREAD

**Discussion by khenderick** [zrepl/zrepl#620] [Q&A] [answered]
I have already a set of snapshots that are replicated to another node, and I want to use zrepl to automate everything.

Assume this current situation:
```
Node A:
foo/bar@5
foo/bar@6

Node B:
foo/bar@3
foo/bar@4
foo/bar@5
foo/bar@6
```

Assuming I've renamed the snapshots so they actually follow the naming pattern used by zrepl (e.g. `zrepl_20220711_180000_000`). How can I make sure zrepl doesn't get confused by the amount of snapshots already existing on both sides. The docs mention the bookmark `#zrepl_replication_cursor`. Does it mean that I can just add the bookmark `#zrepl_replication_cursor` to `foo/bar@6` on both ends?

**Comment by problame** [comment, marked as answer]
1. There's no concept of ownership of snapshots in zrepl. It just replicates all the snapshots it sees. The `prefix` only concerns snaphotting, and your keep rules also apply to all snapshots, not just the ones created by zrepl. Please read the documentation carefully. This PR expands it a little bit, maybe take a look there https://github.com/zrepl/zrepl/pull/614 
2. You don't need to create the replication cursor bookmark. Zrepl will examine both sides, figure out the correspondence, and start replicating incrementally with what is already there. It will just work.

Also, see https://github.com/zrepl/zrepl/discussions/567

**Comment by khenderick** [comment]
Ok, makes sense now, thanks for the info. I also noticed I should have added this to the discussions
